analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/DeliveringDebugFeaturesInProduction.cs

```diff
@@ -43,6 +43,12 @@ public sealed class DeliveringDebugFeaturesInProduction : DeliveringDebugFeature
                 context.Node.FirstAncestorOrSelf<StatementSyntax>() is { } invocationStatement
                 && invocationStatement.Ancestors().Any(node => IsDevelopmentCheck(node, context.SemanticModel));
 
+        protected override TrackerBase<SyntaxKind, InvocationContext>.Condition IsInDevelopmentContext() =>
+            context => context.Node
+                              .Ancestors()
+                              .OfType<ClassDeclarationSyntax>()
+                              .Any(declaration => declaration.Identifier.Text == StartupDevelopment);
```

Comment:
Same as VB - could be `x`, as the `OfType` above is quite clear about the type
```suggestion
                              .Any(x => x.Identifier.Text == StartupDevelopment);
```